src/libs/TransactionUtils.ts

```diff
@@ -0,0 +1,386 @@
+        ...(Object.hasOwn(transactionChanges, 'currency') && {currency: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'merchant') && {merchant: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'waypoints') && {waypoints: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'billable') && {billable: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'category') && {category: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+        ...(Object.hasOwn(transactionChanges, 'tag') && {tag: CONST.RED_BRICK_ROAD_PENDING_ACTION.UPDATE}),
+    };
+
+    return updatedTransaction;
+}
+
+/**
+ * Retrieve the particular transaction object given its ID.
+ *
+ * @deprecated Use withOnyx() or Onyx.connect() instead
+ */
+function getTransaction(transactionID: string): Transaction | Record<string, never> {
+    return allTransactions?.[`${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`] ?? {};
+}
+
+/**
+ * Return the comment field (referred to as description in the App) from the transaction.
+ * The comment does not have its modifiedComment counterpart.
+ */
+function getDescription(transaction: Transaction): string {
+    return transaction?.comment?.comment ?? '';
+}
+
+/**
+ * Return the amount field from the transaction, return the modifiedAmount if present.
+ */
+function getAmount(transaction: Transaction, isFromExpenseReport: boolean): number {
+    // IOU requests cannot have negative values but they can be stored as negative values, let's return absolute value
+    if (!isFromExpenseReport) {
+        const amount = transaction?.modifiedAmount ?? 0;
+        if (amount) {
+            return Math.abs(amount);
+        }
+        return Math.abs(transaction?.amount ?? 0);
+    }
+
+    // Expense report case:
+    // The amounts are stored using an opposite sign and negative values can be set,
+    // we need to return an opposite sign than is saved in the transaction object
+    let amount = transaction?.modifiedAmount ?? 0;
+    if (amount) {
+        return -amount;
+    }
+
+    // To avoid -0 being shown, lets only change the sign if the value is other than 0.
+    amount = transaction?.amount ?? 0;
+    return amount ? -amount : 0;
+}
+
+/**
+ * Return the currency field from the transaction, return the modifiedCurrency if present.
+ */
+function getCurrency(transaction: Transaction): string {
+    const currency = transaction?.modifiedCurrency ?? '';
+    if (currency) {
+        return currency;
+    }
+    return transaction?.currency ?? CONST.CURRENCY.USD;
+}
+
+/**
+ * Return the merchant field from the transaction, return the modifiedMerchant if present.
+ */
+function getMerchant(transaction: Transaction): string {
+    return transaction?.modifiedMerchant ?? transaction?.merchant ?? '';
```

Comment:
This caused a deploy blocker, the `??` operator only returns the right side of the operator if `transaction?.modifiedMerchant` is null but the `transaction?.modifiedMerchant` value in this case is `` an empty string so we return an empty string by mistake